### ISSUE TITLE
connecting to password secured remote pilight-server

### ISSUE THREAD

**Issue by Emiks5** [knalli/homebridge-pilight#65]
First of all thanks for this plugin. I would like to connect pilight to homekit. Now I have setup homebridge on a RPi 3B and Pilight (vs7) on a Rpi 2. I am trying to connect to Pilight using your plugin. It works when I have not set a username-password for the pilight webserver, but I don't know how to connect, when the pilight-webserver is protected by username-password. Would this be possible? And if yes how?
Thanks

**Comment by knalli** [comment]
Hi @Emiks5 

Well, since I moved away my stuff from Homebridge (I replaced my devices with HomeKit-compatible ones), I must admit I am not aware of the compatibility of the more recent versions of pilight. Also I do not know how this behaves with the webgui username/password (I guess you are talking about [this here](https://manual.pilight.org/configuration/settings.html#webserver-authentication)). Maybe this requires an additional authentication step in [this area](https://github.com/knalli/homebridge-pilight/blob/5e4e58dbff058d200246cdc7d7126de9c0d5d67a/lib/ws/WebSocketConnection.js#L61-L64). 

As I haven't revisit this for some time, I am not sure whether the websocket is still the "ideal" communication way.

So, tl;dr: No, that's not possible right now.

**Comment by Emiks5** [comment]
Well thanks anyway. I am now trying to restrict access to the pilight server via the whitelist in the pilight confisg. Not successfull yet though.
